### PR TITLE
baresip 1.1.0

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- previous effort #68868

### CODE REVIEW

File: Formula/baresip.rb [hunk 1]
```diff
@@ -12,21 +13,36 @@ class Baresip < Formula
     sha256 high_sierra: "b99e262d153eb3414c2a6fe813be98e78f71da205d66ede0ec799d1e07f0341a"
   end
 
-  depends_on "libre"
-  depends_on "librem"
+  conflicts_with "libre", because: "both install `libre` lib"
+  conflicts_with "librem", because: "both install `librem` lib"
```
Comment (SMillerDev) [inline Formula/baresip.rb:17]: Why are we vendoring these in the formula now?

Comment (chenrui333) [reply Formula/baresip.rb:17]: They have forked the projects.
https://github.com/baresip/rem/
https://github.com/baresip/re

Comment (carlocab) [reply Formula/baresip.rb:17]: Does it not work with the upstream versions?

Comment (chenrui333) [reply Formula/baresip.rb:17]: Unfortunately, it did not work without forked versions.

**Review by iMichka — DISMISSED** [review]
I asked upstream for clarification: https://github.com/baresip/re/issues/190

**Comment by iMichka** [comment]
@chenrui333 and @carlocab please have a look at https://github.com/baresip/re/issues/190

Wondering what the best course of action is. I am trying to get a little bit more information: https://github.com/baresip/re/discussions/191#discussioncomment-1882532

**Comment by carlocab** [comment]
Given the response at https://github.com/baresip/re/discussions/191

I suggest we do one of:

1. Link these new libraries statically; or
2. Package these new libraries separately (keg-only?) and use them as dependencies for baresip

I strongly prefer to avoid doing anything that requires adding `conflicts_with` to any formulae. Linking statically is probably not ideal if we want to update these libraries separately from baresip itself.

**Comment by iMichka** [comment]
Looking at repology, it's already a huge mess now: https://repology.org/project/libre/versions, https://repology.org/project/librem/versions
You have a mix of upstream repositories for both packages.

But it looks like the `baresip` ones are slowly becoming the new upstream. We might just update both formulae to https://github.com/baresip/re and https://github.com/baresip/rem at this point

**Comment by chenrui333** [comment]
> But it looks like the baresip ones are slowly becoming the new upstream. We might just update both formulae to https://github.com/baresip/re and https://github.com/baresip/rem at this point

Honestly I feel the same.

**Comment by iMichka** [comment]
I have opened a PR: #92948

**Comment by iMichka** [comment]
You should be able to rebase this and use the updated formulae from  #92948

**Comment by chenrui333** [comment]
test failure now
```
  ==> /usr/local/Cellar/baresip/1.1.0/bin/baresip -f /private/tmp/baresip-test-20220113-71987-1w7y929/.baresip -t
  baresip v1.1.0 Copyright (C) 2010 - 2021 Alfred E. Heggestad et al.
  /usr/local/Cellar/baresip/1.1.0/bin/baresip: option requires an argument -- t
  Usage: baresip [options]
```

**Comment by cho-m** [comment]
Related option change causing test failure: https://github.com/baresip/baresip/commit/13d1c9170593dd05d288a0fdfb75dceb622937a4

Previous option: **`-t`               Test and exit**
New option: **`-t <sec>`         Quit after <sec> seconds**

Probably can add a small number after arg.

**Comment by cho-m** [comment]
Existing OS all passed. ARM and Linux still fail due to not finding `librem` headers, which may be due to non-system Homebrew prefix, e.g.
```
src/auframe.c:9:10: fatal error: 'rem.h' file not found
#include <rem.h>
         ^~~~~~~
```

It may be possible to manually set this with something like `LIBREM_PATH` as argument to `make`
```
LIBREM_PATH=#{Formula["librem"].opt_prefix}
```

**Comment by carlocab** [comment]
> Existing OS all passed. ARM and Linux still fail due to not finding `librem` headers, which may be due to non-system Homebrew prefix, e.g.
> 
> ```
> src/auframe.c:9:10: fatal error: 'rem.h' file not found
> #include <rem.h>
>          ^~~~~~~
> ```
> 
> It may be possible to manually set this with something like `LIBREM_PATH` as argument to `make`
> 
> ```
> LIBREM_PATH=#{Formula["librem"].opt_prefix}
> ```

This fixed the build on ARM, but not Linux. Trying to make the build verbose so we have a better idea of what's happening.

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
Ah, this needs an Objective-C(++?) compiler. Not sure we have one of those on Linux.

**Comment by cho-m** [comment]
I think we need to update `make` options for Linux build since AVCapture/CoreAudio/etc are Apple-specific. Need to use alternatives.

Can be done in separate PR since it will require figuring out Linux-only dependencies.

**Comment by carlocab** [comment]
Here's what Arch does: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=baresip

**Pull request closed without merging**